mbedtls/src/rng/hmac_drbg.rs

```diff
@@ -9,51 +9,70 @@
 use mbedtls_sys::types::raw_types::{c_int, c_uchar, c_void};
 use mbedtls_sys::types::size_t;
 pub use mbedtls_sys::HMAC_DRBG_RESEED_INTERVAL as RESEED_INTERVAL;
-use mbedtls_sys::{
-    hmac_drbg_random, hmac_drbg_reseed, hmac_drbg_seed, hmac_drbg_seed_buf,
-    hmac_drbg_set_prediction_resistance, hmac_drbg_update, HMAC_DRBG_PR_OFF, HMAC_DRBG_PR_ON,
-};
+use mbedtls_sys::*;
 
-use super::{EntropyCallback, RngCallback};
+use crate::rng::{EntropyCallback, RngCallback, RngCallbackMut};
 use crate::error::{IntoResult, Result};
 use crate::hash::MdInfo;
+use std::sync::Arc;
 
-define!(
-    #[c_ty(hmac_drbg_context)]
-    struct HmacDrbg<'entropy>;
-    const init: fn() -> Self = hmac_drbg_init;
-    const drop: fn(&mut Self) = hmac_drbg_free;
-);
+#[allow(dead_code)]
+pub struct HmacDrbg {
+    // Moving data causes dangling pointers: https://github.com/ARMmbed/mbedtls/issues/2147
+    // Storing data in heap and forcing rust move to only move the pointer (box) referencing it.
+    // The move will be faster. Access to data will be slower due to additional indirection.
+    inner: hmac_drbg_context,
```

Comment:
Please use the define macro

Reply:
I need an additional field so I cannot use the macro.
```
    entropy: Option<Arc<dyn EntropyCallback + 'static>>,
```

Reply:
There are lots of fields added throughout now with your new scheme, so please adjust the macro to make this possible.

Reply:
This I would leave for a later PR if possible.

Reply:
Updated macro to allow members - this fixes hmac_drbg and context.

For the structures that have Box<c_struct> I have a few options - need to discuss with you which one is best.

Reply:
Added one of the options for the Box<> macro, I have a few other ideas if that doesn't look ok